spec/features/create_etd_spec.rb

```diff
@@ -0,0 +1,235 @@
+      <univid>05543256</univid>
+      <sunetid>dkelley</sunetid>
+      <name>Kelley, DeForest</name>
+      <career code="MED">Medicine</career>
+      <program code="MED">Medical</program>
+      <plan code="ANT">Neurology</plan>
+      <degree>PHD</degree>
+    </DISSERTATION>
+    XML
+  end
+
+  # may need to increase this: file uploads and submission to registrar
+  # Capybara.default_max_wait_time = 5
+
+  # See https://github.com/sul-dlss/hydra_etd/wiki/End-to-End-Testing-Procedure
+  it do
+    # registrar creates ETD in hydra_etd application by posting xml
+    resp_body = simulate_registrar_post(xml_from_registrar)
+    prefixed_druid = resp_body.split.first
+    expect(prefixed_druid).to start_with('druid:')
+    puts "dissertation id is #{dissertation_id}"
+    puts "druid is #{prefixed_druid}"
+
+    etd_submit_url = "https://#{etd_base_url}/submit/#{prefixed_druid}"
+    puts "etd submit url: #{etd_submit_url}" # helpful for debugging
+    authenticate!(start_url: etd_submit_url,
+                  expected_text: "Dissertation ID : #{dissertation_id}")
+    visit etd_submit_url
+
+    # verify citation details
+    expect(page.find('#pbCitationDetails')['style']).to eq '' # citation details not yet verified
+    expect(page).to have_content(dissertation_id)
+    expect(page).to have_content('Kelley, DeForest')
+    expect(page).to have_content(dissertation_title)
+    check('confirmCitationDetails')
+    # a checked box in the progress section is a background image
+    expect(page.find('#pbCitationDetails')['style']).to match(/background-image/)
+
+    # provide abstract
+    expect(page.find('#pbAbstractProvided')['style']).to eq '' # abstract not yet provided
+    abstract_text = 'this is the abstract text'
+    within '#submissionSteps' do
+      step_list = all('div.step')
+      within step_list[1] do
+        find('div#textareaAbstract').click
+        fill_in 'textareaAbstract_edit', with: abstract_text
+        click_button 'Save'
+      end
+    end
+    # a checked box in the progress section is a background image and has class .progressItemChecked
+    expect(page).to have_content(abstract_text)
+    expect(page.find('#pbAbstractProvided')['style']).to match(/background-image/)
+
+    # the hydra_etd app has all the <input type=file> tags at the bottom of the page, disabled,
+    #   and when uploading files, we have to attach the file to the right one of these elements
+    #   I think this may be an artifact of the js framework it usees, prototype
+    file_upload_elements = all('input[type=file]', visible: false)
+
+    # upload dissertation PDF
+    filename = 'etd_dissertation.pdf'
+    expect(page.find('#pbDissertationUploaded')['style']).to eq '' # dissertation PDF not yet provided
+    expect(page).not_to have_content(filename)
+    dissertation_pdf_upload_input = file_upload_elements.first
+    dissertation_pdf_upload_input.attach_file("spec/fixtures/#{filename}")
+    sleep(3) # wait for upload
```

Comment:
This shouldn’t be required with Capybara. It waits on the next assertion until it’s true or there is a timeout. Though I think you need to flip the assertion to be a capybara find.

Reply:
I will double-check -- this may have been an oversight; if not I will leave a better comment as to why we need it.

Reply:
removed -- it was an oversight that I spotted it.